src/requests/getCloudProvider.ts

```diff
@@ -0,0 +1,45 @@
+import axios, { AxiosResponse } from "axios";
+import version from "../utils/version.js";
+import {
+    CloudProviderIdentifier,
+    CloudProviderMapping,
+} from "../models/cloudProviderIdentifier.js";
+import { StatusOk } from "../requests/models.js";
+import { getAuthToken } from "../utils/accounts.js";
+import { BACKEND_ENDPOINT } from "../constants.js";
+import { GENEZIO_NOT_AUTH_ERROR_MSG, UserError } from "../errors.js";
+
+/** This function gets the cloud provider for a project before deployment
+ * It returns the projects cloud provider for existing projects
+ * and the default cloud provider for new projects
+ *
+ * This request should never fail, but because we do not want to cancel the deployment
+ * because of an error here, we provide a default value if anything goes wrong.
+ *
+ * @param name - The name of the project
+ * @returns The cloud provider identifier
+ * @throws UserError if the user is not authenticated
+ */
+export async function getCloudProvider(name: string): Promise<CloudProviderIdentifier> {
+    // Sending an axios request without the interceptor to silently fail if the project is not found.
+    const uninterceptedAxiosInstance = axios.create();
+    const authToken = await getAuthToken();
+
+    if (!authToken) {
+        throw new UserError(GENEZIO_NOT_AUTH_ERROR_MSG);
+    }
+
+    const response: AxiosResponse<StatusOk<{ cloudProvider: CloudProviderIdentifier }>> =
+        await uninterceptedAxiosInstance({
+            method: "GET",
+            url: `${BACKEND_ENDPOINT}/projects/cloud-provider/${name}`,
+            headers: {
+                Authorization: `Bearer ${authToken}`,
+                "Accept-Version": `genezio-cli/${version}`,
+            },
+        });
```

Comment:
I think an error is automatically thrown when the code is not `2**` or `3**`. This will not silently fail, it will throw an unexpected error if the HTTP call is not successful.

Reply:
you are right. i will add a catch statement returning the default value